internal/codec/measurement_test.go

```diff
@@ -0,0 +1,71 @@
+package codec
+
+import (
+	"io"
+	"testing"
+	"time"
+)
+
+func TestMeasureBitRateStatic(t *testing.T) {
+	r, w := io.Pipe()
+	dur := time.Second * 5
+	dataSize := 1000
+	var precision float64 = 8 // 1 byte
+
+	go func() {
+		data := make([]byte, dataSize)
+		for i := time.Second; i < dur; i += time.Second {
```

Comment:
Timing of actual goroutine execution is not guaranteed to be very immediate.
I think it's more stable to put `MeasureBitRate` into routine and iterate `Write` in main routine until `MeasureBitRate` actually ending.

Reply:
Ah, that's a good point. I'll move it to the main routine, and probably add some synchronizations to make it more accurate.